### DISCUSSION TITLE
urllib.urequest

### DISCUSSION THREAD

**Discussion by priisdk** [micropython/micropython#9384] [General]
I'm trying to control my ptz-webcam from esp32 with micropython.
I send  commands like the following to the camera:

```py
urllib.urequest.urlopen("http://"+IPcamadresseexternal+"/command/ptzf.cgi?AbsolutePanTilt="+panhex+','+tilthex+','+str(speed)).read()#pan, tilt, speed
```

This command works perfectly from 'normal' python on my PC, with urllib.request.urlopen in stead of urllib.urequest.urlopen.

It also seems to work on the esp32 (the camera reacts as expected) but I get the following error message:

```
Traceback (most recent call last):
  File "<stdin>", line 168, in <module>
  File "urequests.py", line 116, in get
  File "urequests.py", line 87, in request
IndexError: list index out of range
```

Any help to get rid of this error message would be appreciated.

Regards,
Poul Riis
Denmark

(Edit by jimmo to add code formatting)

**Comment by jimmo** [comment]
@priisdk I'm not quite sure how your stack trace is for `urequests.py` but you're saying you're using `urllib.request`.

I recommend using urequests anyway.

```
import urequests
print(urequests.get("http://"+IPcamadresseexternal+"/command/ptzf.cgi?AbsolutePanTilt="+panhex+','+tilthex+','+str(speed)).content())
```

The error you're seeing looks like it's being caused by an invalid response from the server instead of the expected `HTTP 200 OK` line.

To debug this, can you copy https://github.com/micropython/micropython-lib/blob/master/python-ecosys/urequests/urequests.py to the filesystem of your device (this will replace the built-in version) and try again. You have an older version, and this newer version has a check for this and will raise a `BadStatusLine` exception. Alternatively try flashing the nightly/unstable build of micropython instead of the release version, this will also have the new version of urequests.

It's also weird that to move the camera involves a GET request. GET requests are not supposed to cause state changes. Would make more sense as a POST.

Btw, please see https://github.com/orgs/micropython/discussions/9111 for instructions on code formatting.

**Reply by priisdk** [reply]
I'm sorry about this - it is urequests, not urllib.urequest that causes the error. I have to examine...
